teos/src/main.rs

```diff
@@ -265,11 +266,28 @@ async fn main() {
         internal_rpc_api_uri,
         shutdown_signal_http,
     ));
+    // Add Tor Onion Service for public API
+    if conf.tor_support {
+        log::info!("Starting up hidden tor service");
+        let tor_control_port = conf.tor_control_port;
+        let api_port = conf.api_port;
+        let onion_port = conf.onion_hidden_service_port;
+
+        task::spawn(async move {
+            match tor::expose_onion_service(tor_control_port, api_port, onion_port).await {
+                Ok(_) => {}
+                Err(e) => {
+                    log::error!("{}", e);
+                }
+            };
+        });
```

Comment:
I'd say you want to unwrap here since, if the tor service cannot be set up but the user turned the feature on, the system should not bootstrap.

We hay want to change this from just erroring to something more UX friendly in the future (e.g. make main return a displayable error instead of just unwrapping this results).